# Exploratory Data Analysis of IBM argument mining dataset
The dataset(CE-EMNLP-2015.V3) can be accessed [here](http://www.research.ibm.com/haifa/dept/vst/mlta_data.shtml)

This dataset contains: 2294 labeled claims and 4690 labeled evidence for 58 different topics. Labeled data published by Rinott et al. EMNLP-2015. This data is an extension of the CE-ACL-2014 data.

Detailed description of this dataset can be found [here](https://www.research.ibm.com/haifa/dept/vst/papers/Evidence2015.pdf)

In [76]:
import numpy as np
import pandas as pd
import string
import nltk

# Reading in data

### articles
num: 547

Note: There're 5013 texts within articles folder, so this dataframe only contains articles that are labeled as one of the 58 topics

In [2]:
artDF = pd.read_table("CE-EMNLP-2015.V3/articles.txt")

In [3]:
artDF[:5]

Topic  \
0  This house believes that the sale of violent v...   
1  This house believes that the sale of violent v...   
2  This house believes that the sale of violent v...   
3  This house believes that the sale of violent v...   
4  This house believes that the sale of violent v...   

                              Title  article Id  
0  Video game content rating system         567  
1             Grand Theft Childhood         524  
2                   School violence         542  
3          Video game controversies           1  
4             Nonviolent video game         503

### claims
num: 2294

In [4]:
claimDF = pd.read_table("CE-EMNLP-2015.V3/claims.txt")

In [5]:
print(claimDF['Claim original text'][0])
print(claimDF['Claim corrected version'][0])

Exposure to violent video games causes at least a temporary increase in aggression and this exposure correlates with aggression in the real world
exposure to violent video games causes at least a temporary increase in aggression and that this exposure correlates with aggression in the real world


In [6]:
#claimDF['Claim original text'].apply(lambda x: x.strip(' '))
claimDF[:4]

Topic  \
0  This house believes that the sale of violent v...   
1  This house believes that the sale of violent v...   
2  This house believes that the sale of violent v...   
3  This house believes that the sale of violent v...   

                                 Claim original text  \
0  Exposure to violent video games causes at leas...   
1  video game violence is not related to serious ...   
2  some violent video games may actually have a p...   
3  exposure to violent video games causes both sh...   

                             Claim corrected version  
0  exposure to violent video games causes at leas...  
1  video game violence is not related to serious ...  
2  some violent video games may actually have a p...  
3  exposure to violent video games causes both sh...

### evidence
num: 4692

In [7]:
eviDF = pd.read_table("CE-EMNLP-2015.V3/evidence.txt", names = ['Topic', 'Claim', 'Evidence', 'Evidence Type'])

In [8]:
eviDF[:5]

Topic  \
0  This house believes that the sale of violent v...   
1  This house believes that the sale of violent v...   
2  This house believes that the sale of violent v...   
3  This house believes that the sale of violent v...   
4  This house believes that the sale of violent v...   

                                               Claim  \
0  Exposure to violent video games causes at leas...   
1  Exposure to violent video games causes at leas...   
2  Exposure to violent video games causes at leas...   
3  Exposure to violent video games causes at leas...   
4  Exposure to violent video games causes at leas...   

                                            Evidence Evidence Type  
0  A 2001 study found that exposure to violent vi...       [STUDY]  
1  The most recent large scale meta-anlysis-- exa...       [STUDY]  
2  One study did find an increase in reports of b...       [STUDY]  
3  when one combines all relevant empirical studi...       [STUDY]  
4  Research published in 2001 suggested that viol...       [STUDY]

In [9]:
len(eviDF)

4692

### motions
num: 58

In [10]:
moDF = pd.read_table("CE-EMNLP-2015.V3/motions.txt")

In [11]:
len(moDF)

58

In [12]:
moDF[:5]

Topic id                                              Topic        Data-set
0         1  This house believes that the sale of violent v...  train and test
1        21  This house supports the one-child policy of th...  train and test
2        61  This house would permit the use of performance...  train and test
3        81  This house would make physical education compu...  train and test
4       101  This house believes in the use of affirmative ...  train and test

## Organizing data into ready-to-train format

functions

In [13]:
def clean(x):
    return x.lower().strip(' ').translate(str.maketrans('','',string.punctuation))

In [14]:
def read_text(x):
    t = ''
    filename = "CE-EMNLP-2015.V3/articles/clean_"+str(x)+".txt"
    for line in open(filename, 'r'):
        t += line
    return t

def find_claim(x, dic):
    # finding claims from dictionary{article id: [claims]}
    try:
        return dic[x]
    except:
        return []

clean up claimDF

In [15]:
claimDF['original_rmv_punc'] = claimDF['Claim original text'].apply(lambda x: clean(x))
#claimDF['Claim corrected version'] = claimDF['Claim corrected version'].apply(lambda x: clean(x))

In [16]:
claimDF

Topic  \
0     This house believes that the sale of violent v...   
1     This house believes that the sale of violent v...   
2     This house believes that the sale of violent v...   
3     This house believes that the sale of violent v...   
4     This house believes that the sale of violent v...   
5     This house believes that the sale of violent v...   
6     This house believes that the sale of violent v...   
7     This house believes that the sale of violent v...   
8     This house believes that the sale of violent v...   
9     This house believes that the sale of violent v...   
10    This house believes that the sale of violent v...   
11    This house believes that the sale of violent v...   
12    This house believes that the sale of violent v...   
13    This house believes that the sale of violent v...   
14    This house believes that the sale of violent v...   
15    This house believes that the sale of violent v...   
16    This house believes that the sale of violent v...   
17    This house believes that the sale of violent v...   
18    This house believes that the sale of violent v...   
19    This house believes that the sale of violent v...   
20    This house believes that the sale of violent v...   
21    This house believes that the sale of violent v...   
22    This house believes that the sale of violent v...   
23    This house believes that the sale of violent v...   
24    This house believes that the sale of violent v...   
25    This house believes that the sale of violent v...   
26    This house believes that the sale of violent v...   
27    This house believes that the sale of violent v...   
28    This house believes that the sale of violent v...   
29    This house believes that the sale of violent v...   
...                                                 ...   
2264  This house believes social deprivation causes ...   
2265  This house believes social deprivation causes ...   
2266  This house believes social deprivation causes ...   
2267  This house believes social deprivation causes ...   
2268  This house believes social deprivation causes ...   
2269  This house believes social deprivation causes ...   
2270  This house believes social deprivation causes ...   
2271  This house believes social deprivation causes ...   
2272  This house believes social deprivation causes ...   
2273  This house believes social deprivation causes ...   
2274  This house believes social deprivation causes ...   
2275  This house believes social deprivation causes ...   
2276  This house believes social deprivation causes ...   
2277  This house would enforce term limits on the le...   
2278  This house would enforce term limits on the le...   
2279  This house would enforce term limits on the le...   
2280  This house would enforce term limits on the le...   
2281  This house would enforce term limits on the le...   
2282  This house would enforce term limits on the le...   
2283  This house would enforce term limits on the le...   
2284  This house would enforce term limits on the le...   
2285  This house would enforce term limits on the le...   
2286  This house would enforce term limits on the le...   
2287  This house would enforce term limits on the le...   
2288  This house would enforce term limits on the le...   
2289  This house would enforce term limits on the le...   
2290  This house would enforce term limits on the le...   
2291  This house would enforce term limits on the le...   
2292  This house would enforce term limits on the le...   
2293  This house would enforce term limits on the le...   

                                    Claim original text  \
0     Exposure to violent video games causes at leas...   
1     video game violence is not related to serious ...   
2     some violent video games may actually have a p...   
3     exposure to violent video games causes both sh...   
4     Violent video games increase the violent tende...   
5     No conclusive link was found be

Get original text of these articles

In [17]:
artDF['text'] = artDF['article Id'].apply(lambda x: read_text(x))

In [18]:
artDF[:4]

Topic  \
0  This house believes that the sale of violent v...   
1  This house believes that the sale of violent v...   
2  This house believes that the sale of violent v...   
3  This house believes that the sale of violent v...   

                              Title  article Id  \
0  Video game content rating system         567   
1             Grand Theft Childhood         524   
2                   School violence         542   
3          Video game controversies           1   

                                                text  
0  A video game content rating system is a system...  
1  Grand Theft Childhood: The Surprising Truth Ab...  
2  School violence is widely held to have become ...  
3  . \n\nControversies over video games often cen...

For each article, find the extracted claimes

In [19]:
#claimDF.iloc[0,:]

Topic                      This house believes that the sale of violent v...
Claim original text        Exposure to violent video games causes at leas...
Claim corrected version    exposure to violent video games causes at leas...
original_rmv_punc          exposure to violent video games causes at leas...
Name: 0, dtype: object

In [20]:
dic = {}
notFound = []
for i in range(len(claimDF)):
    topic = claimDF.iloc[i,0]
    claim = str(claimDF.iloc[i,3])
    df = artDF[artDF['Topic']==topic]
    flag = False
    for j in range(len(df)):
        if claim in df.iloc[j,3]:
            flag = True
            try:
                dic[df.iloc[j,2]].append(claim)
            except:
                dic[df.iloc[j,2]] = [claim]
    if flag == False:
        notFound.append((i, topic, claim))

In [21]:
artDF['claims'] = artDF['article Id'].apply(lambda x: find_claim(x,dic))

artDF['num_claim'] = artDF['claims'].apply(lambda x: len(x))

artDF['num_claim'].sum()

726

In [22]:
len(notFound)

1634

In [23]:
notFound

[(0,
  'This house believes that the sale of violent video games to minors should be banned',
  'exposure to violent video games causes at least a temporary increase in aggression and this exposure correlates with aggression in the real world'),
 (4,
  'This house believes that the sale of violent video games to minors should be banned',
  'violent video games increase the violent tendencies among youth'),
 (5,
  'This house believes that the sale of violent video games to minors should be banned',
  'no conclusive link was found between video game usage and violent activity'),
 (6,
  'This house believes that the sale of violent video games to minors should be banned',
  'violent video games are significantly associated with increased aggressive behavior thoughts and affect increased physiological arousal and decreased prosocial helping behavior'),
 (7,
  'This house believes that the sale of violent video games to minors should be banned',
  'video game publishers unethically train c

In [24]:
artDF['sentences'] = artDF['text'].apply(lambda x: [clean(sent) for sent in x.split('.')])
artDF['num_sents'] = artDF['sentences'].apply(lambda x: len(x))

In [25]:
artDF[:4]

Topic  \
0  This house believes that the sale of violent v...   
1  This house believes that the sale of violent v...   
2  This house believes that the sale of violent v...   
3  This house believes that the sale of violent v...   

                              Title  article Id  \
0  Video game content rating system         567   
1             Grand Theft Childhood         524   
2                   School violence         542   
3          Video game controversies           1   

                                                text  \
0  A video game content rating system is a system...   
1  Grand Theft Childhood: The Surprising Truth Ab...   
2  School violence is widely held to have become ...   
3  . \n\nControversies over video games often cen...   

                                              claims  num_claim  \
0  [numerous researchers have proposed potential ...          1   
1  [most children who play violent games do not h...          1   
2                                                 []          0   
3  [video game violence is not related to serious...         10   

                                           sentences  num_sents  
0  [a video game content rating system is a syste...        107  
1  [grand theft childhood the surprising truth ab...         60  
2  [school violence is widely held to have become...        136  
3  [, \n\ncontroversies over video games often ce...        299

In [26]:
#artDF['claims'][0]

In [27]:
#artDF['sentences'][0]

In [28]:
# def determine_label(x):
#      if x

SyntaxError: invalid syntax (<ipython-input-28-06db512e1137>, line 2)

In [29]:
labels = []
for i in range(len(artDF)):
    sents = artDF['sentences'][i]
    claims = artDF['claims'][i]
    label = []
    for sent in sents:
        flag = False
        for claim in claims:
            if claim in clean(sent):
                label.append(1)
                flag = True
                break
        if flag == False:
            label.append(0)
    labels.append(label)

In [30]:
artDF['label'] = labels

In [31]:
sum(artDF['label'][0])

1

In [32]:
artDF['sum_label'] = artDF['label'].apply(lambda x: sum(x))

In [33]:
sum(artDF['sum_label'])

670

Problem: I keep losing claims!!

## What if I don't keep track of the article that the sentences are in, and just determine them altogether.

In [34]:
all_text = ''
for i in artDF['text']:
    all_text += i

In [35]:
all_sents = all_text.split('. ')

In [41]:
claims = claimDF['original_rmv_punc']
labels = []
for sent in all_sents:
    sent = clean(sent)
    flag = False
    for claim in claims:
        if claim in sent:
            labels.append(1)
            flag = True
            break
    if flag == False:
        labels.append(0)

In [37]:
len(all_sents)

75675

In [42]:
len(labels)

75675

In [43]:
sum(labels)

1702

In [44]:
len(claims)

2294

Major problem: when there suppose to be 2294 claims, I'm only able to find 1702 sentences that contain claims using this method.

In [46]:
d = {'Sent': all_sents, 'Contains_claims': labels}
sentDF = pd.DataFrame(d)

In [61]:
sentDF[:5]

Contains_claims                                               Sent  ?
0                0  A video game content rating system is a system...  0
1                0  Most of these systems are associated with and/...  0
2                0  The utility of such ratings has been called in...  0
3                0  Video game content rating systems can be used ...  0
4                0  Rating checking and approval is part of the ga...  0

# Feature extraction
1. ？

In [58]:
def qMark(x):
    try:
        if x[-1] == '?':
            return 1
        else:
            return 0
    except:
        return 0

In [59]:
sentDF['?'] = sentDF['Sent'].apply(lambda x: qMark(x))

2. Belief words

# Text rank/text summarization

In [82]:
import sumy
from sumy.summarizers.text_rank import TextRankSummarizer
from sumy.parsers.html import HtmlParser
from sumy.parsers.plaintext import PlaintextParser
from sumy.nlp.tokenizers import Tokenizer
#from sumy.summarizers.lsa import LsaSummarizer as Summarizer
from sumy.nlp.stemmers import Stemmer
#from sumy.utils import get_stop_words

In [70]:
t = artDF['text'][0]

In [92]:
t

'A video game content rating system is a system used for the classification of video games into suitability-related groups. Most of these systems are associated with and/or sponsored by a government, and are sometimes part of the local motion picture rating system. The utility of such ratings has been called into question by studies that publish findings such as 90% of teenagers claim that their parents "never" check the ratings before allowing them to rent or buy video games,[REF] and as such calls have been made to "fix" the existing rating systems [REF]. Video game content rating systems can be used as the basis for laws that cover the sales of video games to minors, such as in Australia. Rating checking and approval is part of the game localization when they are being prepared for their distribution in other countries or locales. \nThese rating systems have also been used to voluntarily restrict sales of certain video games by stores, such as the German retailer Kaufhof\'s removing

In [90]:
artDF['claims'][0]

['numerous researchers have proposed potential positive effects of video games']

In [84]:
SENTENCES_COUNT = artDF['num_sents'][0]

In [86]:
LANGUAGE = 'English'
parser = PlaintextParser.from_string(t, Tokenizer('English'))
stemmer = Stemmer(LANGUAGE)

summarizer = TextRankSummarizer(stemmer)
summarizer.stop_words = nltk.corpus.stopwords.words('english')
# summarizer = Summarizer(stemmer)
# summarizer.stop_words = get_stop_words(LANGUAGE)

for sentence in summarizer(parser.document, SENTENCES_COUNT):
    print(sentence)

TypeError: 'NoneType' object is not iterable

## raw textrank
https://joshbohde.com/blog/document-summarization

In [87]:
import networkx as nx
import numpy as np
 
from nltk.tokenize.punkt import PunktSentenceTokenizer
from sklearn.feature_extraction.text import TfidfTransformer, CountVectorizer
 
def textrank(document):
    sentence_tokenizer = PunktSentenceTokenizer()
    sentences = sentence_tokenizer.tokenize(document)
 
    bow_matrix = CountVectorizer().fit_transform(sentences)
    normalized = TfidfTransformer().fit_transform(bow_matrix)
 
    similarity_graph = normalized * normalized.T
 
    nx_graph = nx.from_scipy_sparse_matrix(similarity_graph)
    scores = nx.pagerank(nx_graph)
    return sorted(((scores[i],s) for i,s in enumerate(sentences)),
                  reverse=True)

In [91]:
textrank(t)[:10]

[(0.018228815830554752,
  'The (CERO) is the organization that rates video games and PC games (except dating sim, visual novel, and eroge) in Japan with levels of rating that informs the customer of the nature of the product and for what age group it is suitable.'),
 (0.015257808196775656,
  'It is responsible for inspecting and rating the content of films and interactive games.'),
 (0.014390936223200246,
  'A video game content rating system is a system used for the classification of video games into suitability-related groups.'),
 (0.014305087014280488,
  'In July 1994, the IDSA (renamed to the Entertainment Software Association in 2003) presented to Congress its proposal for an industry-controlled rating system; in September of that same year, the Entertainment Software Rating Board (ESRB) was unveiled.'),
 (0.014081788072994273,
  'Video game content rating systems can be used as the basis for laws that cover the sales of video games to minors, such as in Australia.'),
 (0.01383469

In [93]:
a1 = 'Silicon Valley has looked into the future and seen what many of us are seeing – automation and technology are replacing jobs faster than businesses are creating them. What started as a mere murmurings has reverberated louder into an open call from some of the wealthiest businesspeople in the United States including Mark Zuckerberg and Elon Musk for universal basic income.Universal basic income is a stipend paid by the government to individuals simply for being alive. The idea began in Europe and has been gaining momentum in several countries. The principal behind the idea is that it serves as a safety net for unemployment due to market forces such as automation. Several countries have begun experimenting with a basic income including an experiment run in Ontario, Canada, which distributed a stipend of $12, 616 as its basic income. The idea has started to gain traction in America.Hawaii recently became the first state to make a major push towards a universal basic income. As of this May, the Hawaii state legislature adopted HCR 89 calling for the Department of Labor and Industrial Relations and the Department of Business, Economic Development, and Tourism to convene a basic income economic security working group to begin to structure a universal basic income. The relevant text from the resolution gives the group’s focus:WHEREAS, while the United States is the wealthiest nation in the world, many families, individuals, and businesses in Hawaii have been struggling to keep pace with the increasing cost of living as economic inequality widens the gap between a few top earners and the middle and lower class, the latter of which has seen its overall share of income decline in recent decades; and WHEREAS, efforts to increase wages, benefits, and working conditions are important steps to assist local families in the short-term, but a paradigm shift in policy will soon be necessary as automation, innovation, and disruption begin to rapidly worsen economic inequality by displacing significant numbers of jobs in Hawaii’s transportation, food service, tourism, retail, medical, legal, insurance, and other sectors …Hawaii state representative Chris Lee led the push towards a basic income and predicted that Hawaii’s future economic troubles due to Silicon Valley innovations like self-driving cars and Airbnb to be potentially hazardous to Hawaii’s tourist-dependent economy. Though this seems to be an anticipatory measure, however; as of April 2017, Hawaii’s unemployment rate was at a ten year low. This fact didn’t stop the Democrat dominated state legislature from approving the measure.'


In [95]:
a1 += "In other parts of the country, calls in for universal basic income are growing. In California, Silicon Valley has made well known their support for a basic income citing the necessity of social security from the jobs they will take away with their businesses. In the media, a cursory search of liberal publications reveals a bolder comfort with an idea of providing a stipend to people for doing nothing. In Georgia, former documentary assistant turned congressional candidate, Jon Ossoff, went above and beyond agreeing to a $15 minimum wage and instead gave a nod to his California benefactors by saying he would push for a “living wage” in the final debate for the Georgia Sixth Congressional district special election. It’s easy to see the sly rhetorical distinction between “living wage” and “basic income.”There is very little hope that this trend will change without forward-thinking leaders who will refuse to rest on lazy ideas like the universal basic income.Our politicians have been able to see technology replacing jobs for decades now and made little adjustments other than to make more individuals reliant on the government for their economic well being. I echo what I wrote in The Hill last September that automation would require lawmakers to be innovative in their approaches for creating jobs. Many lawmakers and those in the best position to see automation replacing jobs, such as CEO’s in Silicon Valley, it would appear are opting towards a more top-down income redistribution in the form of a universal basic income rather than wrestle with retraining workers, re-calibrating the economy, or encouraging innovation to offset job loss.This is to simply say: universal basic income is a lazy, socialist solution to future unemployment problems. There are dozens of issues with universal basic income, but I will list some of the bigger issues below. As to avoid being an armchair critic who only points out mistakes, I will also offer up some of my own solutions or alternatives to reposition our economy.First, universal basic income stifles economic growth through exorbitant tax rates. To pay for a basic income that would be a living income or sustainable income requires taxation of wage earners well above 50% for the highest and second highest tax brackets. Offsetting high cost of living cities like New York, San Francisco, and Washington, D.C. would necessitate a tax rate unsustainable for most Americans. This leads to either a subsidy for poor Americans, or force many low-income individuals to relocate out of major cities. Needless to say, this exodus hampers innovation.With our current immigration debate, this analysis presumes a comparable rate of illegal immigration and immigration as we have seen in recent years. If history is any indication, the moment a universal basic income comes into law for U.S. citizens, Mayor de Blasio and Governor Cuomo will rush to the front of public debate to argue that universal basic income should be extended to undocumented workers as well since it’s a human right. If that happens, keep your wallets open because that’s going to hurt.Second, universal basic income is a mere proxy for socialism. The plan relies on the assumption that the income will be a social safety net to hedge against unemployment. In a perfect world, individuals would be able to save to offset economic unpredictability. This is an interesting argument to hear coming from a Left who finds the thought of alternatives to social security and Obamacare in the form of private savings account abhorrent. If liberals don’t trust Americans to allocate their money correctly, how can they reasonably suspect that this universal income will be treated differently to realistically serve as a safety net?Finally, a basic universal income only hedges against future unemployment rather than creating opportunities in the present to anticipate economic changes. If automation continues to replace jobs, more Americans will shift from universal basic income being a subsidy to their earned wages towards it being a necessity in unemployment. It’s easy to see how this would spiral the economy into the garbage heap. As workers exit the workforce, fewer workers pay into the pool for distribution meaning that there is less money for the same amount of government handouts. To accommodate the gap, either taxes would have to go up on higher income earners to account for a depleting workforce, or businesses and the government would have to create alternatives to get people to work, or the government would have to require businesses to hire individuals. (Beginning to sound a bit like Atlas Shrugged.) All are bad alternatives. Creating a safety net that promises no ability to put people to work is giving them $1,000, not teaching them to build boats or sail, and asking them to get off an island.There is still time to right the ship. As I said at the outset, nay saying without a solution is as empty as the politicians that will inevitably thrust the universal basic income upon us. There are many avenues to avoiding the calamity from unemployment Silicon Valley predicts from their innovation. I will list several here.First, public schools and innovative STEM charter schools need the resources to teach STEM subjects and trade skills. Understanding how the workforce is being educated and what they are learning are far too often overlooked. For one, a realistic alternative would be to increase trade skills education during high school education while it is cheaper in order to provide career paths for students that do not seek a college degree while equipping them with a skill. It will be a long time before a plumber or electrician can be replaced by a machine. Access to STEM early in education will also improve female participation in STEM fields which has been noticeably absent and will lead to greater market participation in a field that will continue to grow as automation boxes out other fields. Targeting areas that lead to innovation and businesses can allocate resources will put people to work.Please do not misinterpret me to suggest that we need more education. I’d wager there are less than 1,000 people in America using an art history degree for its intended purpose. Instead, this is a proposal to reallocate resources to target types of education where humans will be valuable – trades that machines have difficulty replacing and STEM fields where humans create the machines who replace them.My second proposal is to unburden small businesses. Legislators should be in the business of making it easier for small businesses to operate and comply with government regulation. If state and federal regulation is reduced, small businesses will be best situated to create jobs cheaply. Healthcare constantly being in flux, taxation changes, and legal hurdles make running a small business and creating jobs difficult and in an economy that many predict will desperately need to create employment, government doesn’t need to be a formidable burden in that process.My proposals are not perfect. There are plenty of holes in these proposals and many require intensive policy debates. The debates are worth having because a universal basis income is a road towards a dying economy – settling on a universal basic income is both lazy and simplistic. Without real ingenuity in the way we approach our workforce, we may be doomed to an economy bereft of innovation and liberty."

In [96]:
a1

'Silicon Valley has looked into the future and seen what many of us are seeing – automation and technology are replacing jobs faster than businesses are creating them. What started as a mere murmurings has reverberated louder into an open call from some of the wealthiest businesspeople in the United States including Mark Zuckerberg and Elon Musk for universal basic income.Universal basic income is a stipend paid by the government to individuals simply for being alive. The idea began in Europe and has been gaining momentum in several countries. The principal behind the idea is that it serves as a safety net for unemployment due to market forces such as automation. Several countries have begun experimenting with a basic income including an experiment run in Ontario, Canada, which distributed a stipend of $12, 616 as its basic income. The idea has started to gain traction in America.Hawaii recently became the first state to make a major push towards a universal basic income. As of this Ma

In [97]:
textrank(a1)[:10]

[(0.03032566809019533,
  'The relevant text from the resolution gives the group’s focus:WHEREAS, while the United States is the wealthiest nation in the world, many families, individuals, and businesses in Hawaii have been struggling to keep pace with the increasing cost of living as economic inequality widens the gap between a few top earners and the middle and lower class, the latter of which has seen its overall share of income decline in recent decades; and WHEREAS, efforts to increase wages, benefits, and working conditions are important steps to assist local families in the short-term, but a paradigm shift in policy will soon be necessary as automation, innovation, and disruption begin to rapidly worsen economic inequality by displacing significant numbers of jobs in Hawaii’s transportation, food service, tourism, retail, medical, legal, insurance, and other sectors …Hawaii state representative Chris Lee led the push towards a basic income and predicted that Hawaii’s future econo

SyntaxError: EOL while scanning string literal (<ipython-input-103-2af4bef59de8>, line 1)

In [105]:
b1 = 'Amazon is now the most interesting and important problem in American antitrust law. While it was hardly the first step in the firm’s competitive evolution, Friday’s breathtaking news that the e-commerce giant will acquire Whole Foods brings into sharp focus one of the deepest tensions in our antitrust law as it has evolved under 40 years of conservative orthodoxy. Amazon is enormous, and it is apparently rapacious and remorseless. But at least superficially, it has done nothing in its 20-year life but sell things more cheaply and innovatively than its competitors. That is to say that it’s done just the thing that antitrust law says firms are supposed to do: It has competed on price and quality. The law has therefore proven persistently uninterested in its conduct.While this new merger must undergo federal antitrust review, the Trump administration is unlikely to challenge it. The Obama administration probably wouldn’t have either, and neither would a Clinton administration. That is because, given the state of the law and the attitude of our judiciary, making a case against this deal will seem very challenging indeed. Our antitrust law, as it has come to be, is poorly equipped to deal with a company like Amazon. (Disclosure: Slate is an Amazon affiliate; when you click on an Amazon link from Slate, the magazine gets a cut of the proceeds from whatever you buy.)To many critics, that antitrust is so helpless in the face of a firm so big, acquisitive, and devastating proves how badly it has failed. Its failure, then, reflects an even larger tension, especially on the American left. The struggle is between the haughty pride of our establishment institutions and their apparent impotence against the power and greed from which they are supposed to protect us. The fear and resentment of the poisonous 2016 Democratic presidential primary was driven by nothing less.The better news is that debate over Whole Foods, which seems likely to be pretty lively, could be a bit of a watershed moment for our competition policy. Ideally, recognition might emerge that antitrust as it has evolved, including even its general, simple focus on price competition and protection of consumers, is actually a reasonably tolerable policy. It generates good outcomes most of the time and could get better yet just with more vigorous enforcement and popular support. But hopefully we also will see that a firm like Amazon embodies some things that are wrong with the policy and that it calls for action.That action could actually be taken in the Whole Foods case, and it is not even impossible that the Trump administration might take it. There are a few theories of actionable anticompetitive harm, even on our existing law, by which the government could bring challenge. They will require digging more deeply than the superficial story that Amazon will tell in its own defense, but they won’t require any radical change to the law we’ve got. The new administration ought to take them seriously.Up until now, the mainstream antitrust establishment has been pretty quiet about Amazon, no doubt because its visible conduct fits awkwardly with current ideas of what should be illegal. The Whole Foods acquisition could change that, if for no better reason that it represents a new level of audacity. Though Amazon has acquired hundreds of other firms in the past, some big and many controversial, never before has it bought a significant brick-and-mortar operation. Over the years, as its critics have raised increasingly urgent alarms and elaborated on the firm’s history of fairly nasty aggression, what’s angered them most have been Amazon’s tentative forays into terrestrial stores of its own. They seem to confirm a deliberately predatory, monopolizing strategy: Amazon kills off brick-and-mortar retail competition through desperation prices, only to take over those rivals’ physical presence as soon as they are gone. This new deal is also a big one in absolute dollars. While Whole Foods is by no means the largest American grocer, and the deal’s $14 billion purchase price is not the biggest by any stretch, Whole Foods is among the larger chains in an industry worth upward of a trillion dollars per year. And of course, unlike any of Amazon’s prior targets, Whole Foods is among the best-known names in American retail.But still, an antitrust challenge is quite unlikely. Nowadays, the federal antitrust agencies almost never challenge mergers unless they are “horizontal”—between head-to-head competitors—and they cause very high levels of concentration. This merger is only barely horizontal. The firms’ only overlapping business is online grocery, which so far remains a vanishingly small share of overall grocery. Even if (as seems unlikely) a court found there to be a legally relevant market consisting only of online grocery sales, no serious challenge would be likely. Amazon is a big player in that small space, but Whole Foods is small within it, and there are plenty of other would-be players.Moreover, Amazon will tell a seemingly plausible story to regulators and courts—that it only means to buy a troubled firm in need of a boost and only to support its entry in a space that it could not successfully enter on its own. Amazon has been trying to get a foothold in grocery for a decade, but its only real success has been in England, where it has partnered with the traditional grocer Morrisons.Meanwhile Whole Foods has been stuck in a yearslong doldrums and this year came under dire pressure from Wall Street. Vigorous challenge from mainstream chains and superstores led to a loss of nearly half the firm’s share value over just a few years. Then, just two months ago, a surprise incursion by an activist hedge fund bred immediate speculation that Whole Foods was on the auction block.And yet there is a better way to characterize this deal and at least two good angles for legal attack. Though many Amazon critics fear it calls for some new, reconceptualized antitrust law, it actually seems nicely telling that existing law could deal with the Whole Foods acquisition and with Amazon’s behavior more generally, so long as certain established theories are taken seriously. (Such an approach is also much more politically plausible right now than any holistic, revolutionary change in the underlying law.)First is the argument that Amazon will acquire Whole Foods mainly to kill off an online grocery rival—not now but in the future. This argument would take the deal not so much as “horizontal” and so would begin at a disadvantage. But antitrust protects “dynamic” as well as static competition, and mergers are illegal where they will thwart technological or other developments that pose future competition. Amazon has a history of such maneuvers, as in its notorious, bare-knuckled acquisitions of Zappos and Diapers.com, as well as dozens of smaller acquisitions of otherwise threatening startups. While online grocery is now small, the industry plainly expects it to grow, and while there are currently several would-be entrants, there are specific reasons to expect that this combination will make their futures very, very bleak.(And while it’s not in itself legally relevant, markets have already indicated their confidence in some such anticompetitive outcome. On Friday’s announcement, Amazon’s share price rose. That likely doesn’t reflect investor expectation of efficiency-enhancing integration, because no investors really believe in such things. Wall Street so cynically discounts the motives of executives setting up mergers—something like 70 percent of which fail—that acquiring firms’ shares usually fall when deals are announced. When the acquiring firm’s stock goes up, instead, investors must expect something profitable, and it’s very unlikely they’re just buying the sing-song synergy promises that every CEO makes for every deal.)Better yet, a second theory would challenge Amazon’s conduct overall. A range of its particular actions over a few decades potentially makes out a claim, even under our current antitrust law, of illegal monopolization (in technical terms, a claim under Section 2 of the Sherman Act, rather than a merger challenge under Section 7 of the Clayton Act). The claim would be that this new acquisition and a series of prior acquisitions and other actions have delivered a market position in which Amazon is such a powerful buyer that it can force its suppliers to sell at subcompetitive prices. While monopsony claims are rare and difficult to bring, Amazon is no ordinary antitrust defendant. To prove the claim, the government could first show that Amazon has very large shares of sales in specific products and that those shares reflect actual market power because Amazon has acquired truly, historically unique economies of scope that no retail entrant could meaningfully challenge. The government could then show that these shares were acquired not by healthy, price-competitive vigor (in which case they would not be illegal). Rather, they were gotten through a series of acquisitions that may not have been individually illegal but in combination were monopolistic and by other exclusionary acts. A variety of the latter are well-documented, as, for example, Amazon’s use of its Amazon Marketplace platform to gather analytics on its Marketplace partners, then to enter their markets while using its data advantage and crushing terms to overtake the space.While President Trump’s antitrust leadership is not yet in place, his team appears most likely to consist of well-seasoned lawyer’s lawyers with sterling credentials in antitrust (specifically, Makan Delrahim and, as it is rumored, Joseph Simons). That they are also very conservative does not mean they won’t bring important cases, and this could be one. And the significance of it could not be overstated, not only because of Amazon’s own size and danger but for the broader revitalization it would bring to this crucial policy.'

In [106]:
textrank(b1)

[(0.027132983596283761,
  'Amazon is a big player in that small space, but Whole Foods is small within it, and there are plenty of other would-be players.Moreover, Amazon will tell a seemingly plausible story to regulators and courts—that it only means to buy a troubled firm in need of a boost and only to support its entry in a space that it could not successfully enter on its own.'),
 (0.025796669950298744,
  'But hopefully we also will see that a firm like Amazon embodies some things that are wrong with the policy and that it calls for action.That action could actually be taken in the Whole Foods case, and it is not even impossible that the Trump administration might take it.'),
 (0.023999653562320596,
  'And the significance of it could not be overstated, not only because of Amazon’s own size and danger but for the broader revitalization it would bring to this crucial policy.'),
 (0.022704851496836192,
  'The claim would be that this new acquisition and a series of prior acquisition